Fixed transition counts inclusive/exclusive

Immersor-git opened the pull request:
Fixes #27

Immersor-git committed:
Fixed transition counts inclusive/exclusive

ashkjones commented on src/analysis/AreaOfInterests.java:
```diff
@@ -7,7 +7,7 @@
 
 public class AreaOfInterests {
-    final static String FIXATIONID_INDEX = "FPOGID";
+    final static String FIXATIONID_INDEX = "CNT"; //FPOGID
```
@Immersor-git Could you explain why CNT is used over FPOGID? Not saying its wrong but I just want to understand the code better.

ashkjones reviewed (commented):
<img width="500" alt="image" src="https://github.com/TheD2Lab/gazepoint-data-analysis/assets/4361319/97311979-142f-4cbe-ba44-d46ad91b1b20">

I tried testing with p11's data (arbitrarily chosen). I summed the number of transitions (103976) and found it was higher than the number of fixations (1369) in the DGM.csv. I don't think there should be more transitions than there are fixations.

I understand this is due to multi-AOI fixations, but I think we decided to only use the FPOGX and Y of the fixation entry (not the intermediate data records) to assign the AOI. I think you want to go back to using `allFixations` and not `allGazeData`. I could be mistaken though. Maybe ask Dr. Fu how to proceed.

Immersor-git committed:
Made adjustments to work with fixations rather than allGaze

Immersor-git commented:
> <img alt="image" width="500" src="https://private-user-images.githubusercontent.com/4361319/326998949-97311979-142f-4cbe-ba44-d46ad91b1b20.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1MTgwNDksIm5iZiI6MTcxNDUxNzc0OSwicGF0aCI6Ii80MzYxMzE5LzMyNjk5ODk0OS05NzMxMTk3OS0xNDJmLTRjYmUtYmE0NC1kNDZhZDkxYjFiMjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MzBUMjI1NTQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGJjNzhmZDM5ZWZmMTU3ODQzNzhmZGFhNzBjMDFjODBhM2FiNjAxNWIzMGJjMWI0ZTlhNDFjNWY0ZGUwZTc3YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.8M0A9ZAvKDoghtiAMroY6Z_gkpHONYKiQ1kbwF-xV3g">
> I tried testing with p11's data (arbitrarily chosen). I summed the number of transitions (103976) and found it was higher than the number of fixations (1369) in the DGM.csv. I don't think there should be more transitions than there are fixations.
> 
> I understand this is due to multi-AOI fixations, but I think we decided to only use the FPOGX and Y of the fixation entry (not the intermediate data records) to assign the AOI. I think you want to go back to using `allFixations` and not `allGazeData`. I could be mistaken though. Maybe ask Dr. Fu how to proceed.

I had switched back to allGaze instead of allFixations, as it seemed that a lot of transitions weren't registered when using allFixations. Part of the reason is related to your earlier question on using CNT over FPOGID. It turned out that FPOGID didn't increase every time there was an AOI transition, so a lot of them would be missed the lines were expected to have subsequent FPOGID values. 

I've just pushed another fix for this by counting transitions that have identical or subsequent FPOGID values, and switched back to using allFixations instead of allGaze - the resulting numbers appear much more reasonable now.

![image](https://github.com/TheD2Lab/gazepoint-data-analysis/assets/62318692/e0562196-0ef4-44c4-a0b6-9cec8bb3180f)

Immersor-git committed:
Final update - uses allFixations and subsequent FPOGIDs,.

Immersor-git commented:
@ashkjones 
Pushed on final update - uses FPOGID again, checks for subsequency between FPOGIDs only, and uses allFixations instead of allGaze. This should be the final patch required.
